cranelift/frontend/src/frontend/safepoints.rs

```diff
@@ -82,350 +98,686 @@ impl<T> SlotSizeMap<T> {
     }
 }
 
-impl FunctionBuilder<'_> {
-    /// Insert spills for every value that needs to be in a stack map at every
-    /// safepoint.
-    ///
-    /// We begin with a very simple, imprecise, and overapproximating liveness
-    /// analysis. This considers any use (regardless if that use produces side
-    /// effects or flows into another instruction that produces side effects!)
-    /// of a needs-stack-map value to keep the value live. This allows us to do
-    /// this liveness analysis in a single post-order traversal of the IR,
-    /// without any fixed-point loop. The result of this analysis is the mapping
-    /// from each needs-stack-map value that is live across a safepoint to its
-    /// associated stack slot.
-    ///
-    /// Finally, we spill each of the needs-stack-map values that are live
-    /// across a safepoint to their associated stack slot upon definition, and
-    /// insert reloads from that stack slot at each use of the value.
-    pub(super) fn insert_safepoint_spills(&mut self) {
+/// A set of live values.
+type LiveSet = HashSet<ir::Value>;
```

Comment:
Do we ever iterate over the live-set and produce code based on its order? It'd be good to confirm (with a comment noting here) that we don't, in order to retain deterministic compiler output.

Reply:
The one place we do iterate over them, it is only after we've copied them into a smallvec and sorted it. But I added a note to the doc comment as well.